services/thermal_mgr/thermal_mgr.c

```diff
@@ -41,21 +43,86 @@ void initThermalSystemManager(lm75bd_config_t *config) {
+ *
+ * @param event - Pointer to a thermal_mgr_event_t to add to the queue.
+ */
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+    /* Send an event to the thermal manager queue */    
+    // Check for invalid parameter
+    if( event == NULL ) {
+        return ERR_CODE_INVALID_ARG;
+    }
+    // Check if the queue has been created
+    if( thermalMgrQueueHandle == NULL ) {
+        return ERR_CODE_UNKNOWN;
+    }
+
+    switch( xQueueSend( thermalMgrQueueHandle, (void *) event, 0 ) ) {
+        case pdTRUE:
+            return ERR_CODE_SUCCESS; 
+            break;
+        case errQUEUE_FULL:
+            return ERR_CODE_QUEUE_FULL;    
+            break;
+        default:
+            return ERR_CODE_UNKNOWN;
+    }
 }
-
-void osHandlerLM75BD(void) {
-  /* Implement this function */
+/* 
+ * @brief Handles overtemperature events from temperature sensor. 
+ */
+void osHandlerLM75BD() {    
+    recomputeOTStateFlag = 1;
+    // Reset overtemperature   
+    thermal_mgr_event_t resetEvent;
+    resetEvent.type = THERMAL_MGR_EVENT_MEASURE_TEMP_CMD;  
+
+    thermalMgrSendEvent( &resetEvent );   
 }
 
-static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
-  }
+/*
+ * @brief Task responsible for parsing events in its queue 
+ *
+ * @param pvParameters - Configuration struct for the LM75BD temp sensor 
+ */
+static void thermalMgr(void *pvParameters) {      
+    // XXX: It is assumed we don't have to initialize temperature sensor
+    lm75bd_config_t *tempSensorConfig;  
+    error_code_t errCode;
+    tempSensorConfig = (lm75bd_config_t *) pvParameters;   
+
+    while (1) {  
+        thermal_mgr_event_t *event_recieved;
+        if ( xQueueReceive( thermalMgrQueueHandle, (void *) event_recieved, 0) == pdTRUE ) {    
+            // We got an event from the thermalMgr queue, parse it
+            switch( event_recieved->type ){  
+                case THERMAL_MGR_EVENT_MEASURE_TEMP_CMD:    
+                        // Read temperature 
+                        float temperatureResult = 0.0;   
+                        errCode = readTempLM75BD( tempSensorConfig->devAddr, &temperatureResult );  
+                       
+                        // Only change overtemperature state if we had an interrupt raised recently
+                        if (recomputeOTStateFlag) {  
+
+                            if ( temperatureResult > 80.0 ) { 
+                                overTemperatureDetected();
+                            } else {
+                                safeOperatingConditions();
+                            }  
+                            // We've handled the flag, reset it
+                            recomputeOTStateFlag = 0;
+                        }
+
+                        addTemperatureTelemetry( temperatureResult );
+                    break; 
+                deafult:
```

Comment:
```suggestion
                default:
```

It's weird that the compiler didn't catch this.